### PR TITLE
Replaced Go channel with RabbitMQ queuing

### PR DESCRIPTION
**Pull request by alexdunne**
Workers were previously receiving from a in-memory queue. This has now been moved to a RabbitMQ queue.

### CODE REVIEW

File: cmd/consumer/main.go [hunk 1]
```diff
@@ -75,25 +89,58 @@ func main() {
...
-		log.Fatal(errors.Wrap(err, "opening store db connection"))
+		logger.Fatal("failed to create db connection", zap.Error(err))
 	}
 	defer db.Close()
 
 	logger.Info("creating HN client")
-	hn := hn.New()
+	hnClient := hn.New()
+
+	rmq, err := queue.New(cfg.RabbitMQURL, queueName, logger)
```
Comment (jordanfinners) [inline cmd/consumer/main.go]: not really a fan of the naming here for hnClient or rmq
would prefer `hackerNewsClient` and `queueClient`

File: internal/consumer/worker.go [hunk 2]
```diff
@@ -25,21 +26,23 @@ func NewWorker(logger *zap.Logger, db database.Database, hn hn.Client) *Worker {
 	}
 }
 
-func (w *Worker) run(ctx context.Context, idStream <-chan int, wg *sync.WaitGroup) {
+func (w *Worker) Run(ctx context.Context, message <-chan *queue.Message, wg *sync.WaitGroup) {
```
Comment (jordanfinners) [inline internal/consumer/worker.go:32]: Exported should have comment

File: internal/queue/factory.go [hunk 3]
```diff
@@ -0,0 +1,137 @@
...
+	"github.com/pkg/errors"
+	"github.com/streadway/amqp"
+	"go.uber.org/zap"
+)
+
+type Message struct {
+	ID int `json:"id"`
+}
+
+type Queue interface {
```
Comment (jordanfinners) [inline internal/queue/factory.go:18]: Exported needds comments

File: internal/queue/factory.go [hunk 4]
```diff
@@ -0,0 +1,137 @@
...
+		return nil, errors.Wrap(err, "failed to connect to rabbitmq")
+	}
+
+	amqpChan, err := conn.Channel()
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to create a channel")
+	}
+
+	q, err := amqpChan.QueueDeclare(
+		queueName, // name
```
Comment (jordanfinners) [inline internal/queue/factory.go]: this is very helpful

File: internal/queue/factory.go [hunk 5]
```diff
@@ -0,0 +1,137 @@
...
+		false,        // no-wait
+		nil,          // args
+	)
+	if err != nil {
+		return nil, err
+	}
+
+	messages := make(chan *Message)
+
+	go func() {
```
Comment (jordanfinners) [inline internal/queue/factory.go:112]: I think i'd extract this rather than have an inline function

Comment (alexdunne) [reply internal/queue/factory.go:112]: Any reason for the preference?

**Pull request merged**